source/MaterialXGenOgsXml/OgsFragment.cpp

```diff
@@ -0,0 +1,353 @@
+#include "OgsFragment.h"
+
+#include <MaterialXFormat/XmlIo.h>
+#include <MaterialXGenShader/Util.h>
+#include <MaterialXGenShader/DefaultColorManagementSystem.h>
+#include <MaterialXGenOgsXml/GlslFragmentGenerator.h>
+#include <MaterialXGenShader/GenContext.h>
+#include <MaterialXGenOgsXml/OgsXmlGenerator.h>
+
+#ifdef MATERIALX_BUILD_CROSS
+#include <MaterialXCross/Cross.h>
+#endif
+
+#include <iostream>
+
+namespace MaterialXMaya
+{
+namespace
+{
+// The base class for classes wrapping GLSL fragment generators for use during
+// OgsFragment construction.
+class GlslGeneratorWrapperBase
+{
+    GlslGeneratorWrapperBase() = delete;
+
+  protected:
+    explicit GlslGeneratorWrapperBase(mx::ElementPtr element)
+        : _element(element)
+    {
+        if (!_element)
+            throw mx::Exception("No element specified");
+
+        if (element->asA<mx::ShaderRef>())
+        {
+            _isSurface = true;
+        }
+        else if (!element->asA<mx::Output>())
+        {
+            throw mx::Exception("Invalid element to create fragment for " + element->getName());
+        }
+    }
+
+  protected:
+    void setCommonOptions(mx::GenOptions& genOptions, const mx::ShaderGenerator& generator)
+    {
+        // Use FIS environment lookup for surface shader generation but
+        // disable for texture nodes to avoid additional unneeded XML parameter
+        // generation.    
+        genOptions.hwSpecularEnvironmentMethod =
+            _isSurface ? mx::SPECULAR_ENVIRONMENT_FIS : mx::SPECULAR_ENVIRONMENT_NONE;
+
+        // Set to use no direct lighting
+        genOptions.hwMaxActiveLightSources = 0;
+
+        // Maya images require a texture coordinates to be flipped in V.
+        genOptions.fileTextureVerticalFlip = true;
+        genOptions.hwTransparency = mx::isTransparentSurface(_element, generator);
```

Comment:
This way, the test harness shares many of the generation options with the Maya plugin.